Frontend/nutri-pal/src/app/diary/page.tsx

```diff
@@ -0,0 +1,223 @@
+        const HISTORY_DAYS = BASE_URL_REST_API + 'history/registered_days/' + user.uid + `?year=${year}&month=${month}&day=${day}`;
+        axios.get(HISTORY_DAYS)
+            .then((response) => { setDates(response.data) })
+            .catch((error) => { alert('There has been an error trying to get the dates of the user \n' + error) })
+    }
+
+    // Do a GET request to get the maximum and minimum registered days the user has
+    const getLimitDates = () => {
+        const MAX_MIN_DATES = BASE_URL_REST_API + 'history/maxmindate/' + user.uid;
+        axios.get(MAX_MIN_DATES)
+            .then((response) => { setMaxDate(response.data._max.date.slice(0, 4)); setMinDate(response.data._min.date.slice(0, 4)) })
+            .catch((error) => { alert('There has been an error trying to get the maximum and minimum dates of the user \n' + error) })
+    }
+
+    // After auth make the GET requests to display the cards
+    useEffect(() => {
+        if (getFlag) {
+            getDates()
+            getLimitDates()
+        }
+    }, [getFlag])
+
+    // Function to open and closing the modal
+    const toggleModal = () => {
+        if (openModal) {
+            setOpenModal(false)
+        } else {
+            setOpenModal(true)
+        }
+    }
+
+    // When the input calendar date is changed then reset the values of the selected year and month filter, and then separate the string into year month and date for the GET request
+    useEffect(() => {
+        setFilterYear('')
+        setFilterMonth('')
+        if (filterDate.length > 0) {
+            const temp = filterDate.split('-');
+            getDates(temp[0], temp[1], temp[2]);
+        } else {
+            getDates()
+        }
+    }, [filterDate])
+
+    // When the a year is selected from the filter option then able the month filter option and do the GET request with the query params of just the year
+    useEffect(() => {
+        if (filterYear.length > 0) {
+            setMonthFlag(true)
+            getDates(filterYear)
+        } else {
+            setMonthFlag(false)
+            setFilterMonth('')
+            getDates()
+        }
+    }, [filterYear])
+
+    // When the month is selected from the filter option then do a GET request with they query params of the year and the month
+    useEffect(() => {
+        if (monthFlag) {
+            getDates(filterYear, filterMonth)
+        }
+    }, [filterMonth])
+
+    // When the minDate is received from the GET request when loading the page, then create a array to get all the years from the least to the greatest
+    useEffect(() => {
+        if (minDate.length > 0) {
+            const newRangeYear: any = []
+            for (let minYear = parseInt(minDate); minYear <= parseInt(maxDate); minYear++) {
+                newRangeYear.push(minYear)
+            }
+            setRangeYear(newRangeYear)
+        }
+    }, [minDate])
+
+    return (
+        <>
+            <TopBarMain />
```

Comment:
for style points you could do something like this
1. create a file SomeContainer.tsx
Then within it it can look something like 
```
const SomeContainer({children}) => {
  return <>
    <SomeTopBar />
    {children}
  </>
}
```


2. Then for this diary component
```
return 
<SomeContainer> 
 {whatever children you want}
</SomeContainer>

```